### PR TITLE
Fix MultiClusterSearchYamlTestSuiteIT test failures

### LINKED ISSUE #58345: [CI] MultiClusterSearchYamlTestSuiteIT test failures: _shards.total didn't match expected value
**Build scan**: https://gradle-enterprise.elastic.co/s/tak2dqnsyoy3u

**Repro line**:

```
REPRODUCE WITH: gradlew ':qa:multi-cluster-search:mixedClusterTestRunner' --tests "org.elasticsearch.search.MultiClusterSearchYamlTestSuiteIT.test {yaml=multi_cluster/*}" \
  -Dtests.seed=E65AA623CD0A6CA2 \
  -Dtests.security.manager=true \
  -Dtests.locale=ms-MY \
  -Dtests.timezone=America/Buenos_Aires \
  -Druntime.java=8 \
  -Dtests.rest.suite=multi_cluster
```

**Reproduces locally?**: Yes, but so far only with specific seeds. 

**Applicable branches**: 7.x, master

**Failure history**:
One release build and three PRs in the last two days. Nothing earlier. See [gradle enterprise test history](https://gradle-enterprise.elastic.co/s/gqpvplz7vhbkk/tests/:qa:multi-cluster-search:mixedClusterTestRunner/org.elasticsearch.search.MultiClusterSearchYamlTestSuiteIT/test%20%7Byaml=multi_cluster%2F80_index_name_agg%2FTest%20that%20remote%20indices%20are%20maintained%20when%20we%20aggregate%20on%20_index%7D#1)

**Failure excerpt**:

```
java.lang.AssertionError: Failure at [multi_cluster/80_index_name_agg:45]: _shards.total didn't match expected value: |  
-- | --
  | _shards.total: expected Integer [2] but was Integer [3] |  
  | at __randomizedtesting.SeedInfo.seed([11FAA49E7D1823A6:99AE9B44D3E44E5E]:0) |  
  | at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:405) |  
  | at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:382) |  
  | at sun.reflect.GeneratedMethodAccessor13.invoke(Unknown Source) |  
  | at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) |  
  | at java.lang.reflect.Method.invoke(Method.java:498) |  
  | at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750) |  
  | at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938) |  
  | at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974) |  
  | at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988) |  
  | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  | at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49) |  
  | at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45) |  
  | at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48) |  
  | at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64) |  
  | at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47) |  
  | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  | at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368) |  
  | at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817) |  
  | at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468) |  
  | at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947) |  
  | at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832) |  
  | at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883) |  
  | at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894) |  
  | at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45) |  
  | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  | at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41) |  
  | at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40) |  
  | at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40) |  
  | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  | at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53) |  
  | at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47) |  
  | at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64) |  
  | at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54) |  
  | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  | at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368) |  
  | at java.lang.Thread.run(Thread.java:748) |  
  |   |  
  | Caused by: |  
  | java.lang.AssertionError: _shards.total didn't match expected value: |  
  | _shards.total: expected Integer [2] but was Integer [3] |  
  | at org.elasticsearch.test.rest.yaml.section.MatchAssertion.doAssert(MatchAssertion.java:93) |  
  | at org.elasticsearch.test.rest.yaml.section.Assertion.execute(Assertion.java:76) |  
  | at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:398) |  
  | ... 37 more
```

Timing-wise, this might have to do with #58298, but I can't say for sure. I haven't had much luck digging into this one.

I will mute the failing tests here.

### PR DESCRIPTION
**Pull request by jimczi**
This commit fixes the multi cluster yaml tests by restoring the number of shards to 1 for the single_doc_index (removed in #57276).

Closes #58345

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/search/aggregations/bucket/terms/LongKeyedBucketOrdsTests.java [hunk 1]
```diff
@@ -104,7 +104,6 @@ private void collectsFromSingleBucketCase(LongKeyedBucketOrds ords) {
         }
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/58353")
```
Comment (cbuescher) [inline server/src/test/java/org/elasticsearch/search/aggregations/bucket/terms/LongKeyedBucketOrdsTests.java:107]: Is this removed by accident here? I don't see the relation to the change in the rest test above. Also the error  reproduction from #58353 still works for me.

File: qa/multi-cluster-search/src/test/resources/rest-api-spec/test/remote_cluster/10_basic.yml [hunk 2]
```diff
@@ -50,6 +50,7 @@
         body:
           settings:
             index:
+              number_of_shards: 1
```
Comment (cbuescher) [inline qa/multi-cluster-search/src/test/resources/rest-api-spec/test/remote_cluster/10_basic.yml:53]: Not sure this change is in the right yaml test. The correspoinding @Awaitsfix for #58345 seems to be in multi-cluster-search/src/test/resources/rest-api-spec/test/multi_cluster/80_index_name_agg.yml

**Review by cbuescher — COMMENTED** [review]
I'm not sure the removed "@Awaitsfix" matches the one added for #58345, could you re-check this? Maybe the change should have gone into another file?

File: qa/multi-cluster-search/src/test/resources/rest-api-spec/test/remote_cluster/10_basic.yml (hunk 2 above)
Comment (jimczi) [reply qa/multi-cluster-search/src/test/resources/rest-api-spec/test/remote_cluster/10_basic.yml:53]: This part defines the remote indices that are used in the tests. See https://github.com/elastic/elasticsearch/blob/master/qa/multi-cluster-search/src/test/resources/rest-api-spec/test/multi_cluster/80_index_name_agg.yml#L36

**Comment by jimczi** [comment]
> I'm not sure the removed "@Awaitsfix" matches the one added for #58345, could you re-check this? Maybe the change should have gone into another file?

I reverted the wrong commit, thanks. It should be good now so can you take another look ?

**Comment by cbuescher** [comment]
I tried the reproduction from #58345 which unfortunately still fails reproducibly for me locally:
```
./gradlew ':qa:multi-cluster-search:mixedClusterTestRunner' --tests "org.elasticsearch.search.MultiClusterSearchYamlTestSuiteIT.test {yaml=multi_cluster/*}" -Dtests.seed=E65AA623CD0A6CA2
```
```
REPRODUCE WITH: ./gradlew ':qa:multi-cluster-search:mixedClusterTestRunner' --tests "org.elasticsearch.search.MultiClusterSearchYamlTestSuiteIT.test {yaml=multi_cluster/10_basic/Search an filtered alias and empty index on the remote cluster}" -Dtests.seed=E65AA623CD0A6CA2 -Dtests.security.manager=true -Dtests.locale=pl -Dtests.timezone=America/Godthab -Druntime.java=11 -Dtests.rest.suite=multi_cluster
  2> java.lang.AssertionError: Failure at [multi_cluster/10_basic:325]: _shards.total didn't match expected value:
                     _shards.total: expected Integer [4] but was Integer [5]
        at __randomizedtesting.SeedInfo.seed([E65AA623CD0A6CA2:6E0E99F963F6015A]:0)
        at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:405)
        at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:382)
```
I'm curious to understand why this doesn't fail CI, ie which part is determined by the random seed here. Maybe the`field_caps_empty_index` queries in the test thats failing still for me now also needs some reverted shard settings (https://github.com/elastic/elasticsearch/blob/master/qa/multi-cluster-search/src/test/resources/rest-api-spec/test/multi_cluster/10_basic.yml#L322)? Haven't looked at what that would affect though.

**Comment by jimczi** [comment]
> I'm curious to understand why this doesn't fail CI, ie which part is determined by the random seed here. Maybe thefield_caps_empty_index queries in the test thats failing still for me now also needs some reverted shard settings (https://github.com/elastic/elasticsearch/blob/master/qa/multi-cluster-search/src/test/resources/rest-api-spec/test/multi_cluster/10_basic.yml#L322)?

The number of shards is randomized in yaml tests so we have to force a number if we check this number on search responses. I pushed https://github.com/elastic/elasticsearch/pull/58359/commits/d4bd6ef5748600747a96482391194494edfca804 to fix the failing test.

**Review by cbuescher — APPROVED** [review]
> The number of shards is randomized in yaml tests so we have to force a number

Makes sense, I thought maybe its using the default 5 shards or something fixed if no setting is provided. Thanks.
LGTM

**Pull request merged**